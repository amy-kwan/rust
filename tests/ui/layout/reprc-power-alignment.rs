//@ check-pass
//@ revisions: aix
//@[aix] compile-flags: --target powerpc64-ibm-aix
//@[aix] needs-llvm-components: powerpc

#![warn(uses_power_alignment)]

#[repr(C)]
pub struct Floats {
    a: f64,  //~ WARNING repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
    b: u8,
    c: f64,
}

pub struct Floats2 {
    a: f64,
    b: u32,
    c: f64,
}

#[repr(C)]
pub struct Floats3 {
    a: f32,  //~ WARNING repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
    b: f32,
    c: i64,
}

#[repr(C)]
pub struct Floats4 {
    a: u64,
    b: u32,
    c: f32,
}

#[repr(C)]
pub struct FloatAgg1 {
    x: Floats, //~ WARNING repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
    y: f64,
}

#[repr(C)]
pub struct FloatAgg2 {
    x: i64,
    y: Floats,
}

#[repr(C)]
pub struct FloatAgg3 {
    x: FloatAgg1, //~ WARNING repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
    y: FloatAgg2,
}

fn main() { }
