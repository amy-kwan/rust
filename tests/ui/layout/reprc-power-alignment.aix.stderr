warning: repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
  --> $DIR/reprc-power-alignment.rs:10:5
   |
LL |     a: f64,
   |     ^^^^^^
   |
note: the lint level is defined here
  --> $DIR/reprc-power-alignment.rs:6:9
   |
LL | #![warn(uses_power_alignment)]
   |         ^^^^^^^^^^^^^^^^^^^^

warning: repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
  --> $DIR/reprc-power-alignment.rs:23:5
   |
LL |     a: f32,
   |     ^^^^^^

warning: repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
  --> $DIR/reprc-power-alignment.rs:37:5
   |
LL |     x: Floats,
   |     ^^^^^^^^^

warning: repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
  --> $DIR/reprc-power-alignment.rs:49:5
   |
LL |     x: FloatAgg1,
   |     ^^^^^^^^^^^^

warning: 4 warnings emitted

